docs/docs/documentation/components/projects.md

```diff
@@ -0,0 +1,375 @@
+
+- Inherit the project's instructions (if defined)
+- Use the project's default agent (if specified)
+- Can access project-level artifacts
+- Can be moved between projects
+- Can be searched and filtered
+
+### Knowledge (Artifacts)
+
+Projects support two types of knowledge artifacts:
+
+1. **Project-Level Knowledge**: Files attached to the project itself (shown in the "Knowledge" section), accessible by all sessions
+2. **Session-Level Artifacts**: Files attached to specific chat sessions within the project
+
+Knowledge artifacts can include:
+- Documents (PDF, DOCX, TXT, MD)
+- Images (PNG, JPG, GIF)
+- Code files
+- Data files (JSON, CSV, YAML)
+- Any other file type supported by the system
+
+## API Endpoints
+
+### Project Management
+
+| Method | Endpoint | Description |
+|--------|----------|-------------|
+| POST | `/api/v1/projects` | Create a new project |
+| GET | `/api/v1/projects` | List all projects for the current user |
+| GET | `/api/v1/projects/{id}` | Get a specific project by ID |
+| PUT | `/api/v1/projects/{id}` | Update project details |
+| DELETE | `/api/v1/projects/{id}` | Soft delete a project |
+
+### Artifact Management
+
+| Method | Endpoint | Description |
+|--------|----------|-------------|
+| GET | `/api/v1/projects/{id}/artifacts` | List all artifacts in a project |
+| POST | `/api/v1/projects/{id}/artifacts` | Add artifacts to a project |
+| DELETE | `/api/v1/projects/{id}/artifacts/{filename}` | Delete an artifact from a project |
+
+### Session Management
+
+| Method | Endpoint | Description |
+|--------|----------|-------------|
+| PATCH | `/api/v1/sessions/{id}/project` | Move a session to a different project |
+| GET | `/api/v1/sessions/search` | Search sessions by name or content |
+| DELETE | `/api/v1/sessions/{id}` | Soft delete a session |
+
+## Configuration
+
+### Enabling Projects
+
+Projects require SQL database persistence to function. Configure persistence in your `shared_config.yaml`:
+
+```yaml
+session_service:
+  type: sql
+  database_url: "sqlite:///./data/sessions.db"
+```
+
+Projects are enabled by default when persistence is configured. To explicitly control the feature:
+
+```yaml
+# Enable projects (default when persistence is enabled)
+projects:
+  enabled: true
+
+# Or disable projects explicitly
+projects:
+  enabled: false
+```
+
+### Feature Flag Control
+
+You can also control projects via feature flags:
+
+```yaml
+frontend_feature_enablement:
+  projects: true  # Enable projects
+  taskLogging: true
+```
+
+:::note[Configuration Priority]
+The feature flag resolution follows this priority:
+1. **Persistence Check**: If persistence is disabled, projects are disabled (non-negotiable)
+2. **Explicit Config**: `projects.enabled` setting
+3. **Feature Flag**: `frontend_feature_enablement.projects` setting
+4. **Default**: Enabled (if persistence is enabled and no explicit disable)
+:::
+
+## Usage Examples
```

Comment:
This section is not needed, this will be UI driven. We can delete up to `Projects and sessions use a soft delete pattern for data preservation`